app/models/user.rb

```diff
@@ -6,9 +6,10 @@ class User < ApplicationRecord
   after_create :profile_creation
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
-  has_many :posts, foreign_key: :author_id
+  has_many :posts, foreign_key: :author_id, dependent: :destroy
+  has_many :comments, foreign_key: :author_id
   has_one :profile
```

Comment:
### Note
**your user_controller isn't here because you merged before so these are for user_controller***
- you don't need a new method, you are creating user using devise remeber
- you dont need to do update either, devise has and update method, and since the profile is connected to it consider updating from there so you can add the email, city etc fields to the  update as well
- something like name should be on the user, that way they user provides a name at registration

Reply:
The name is registered automatically based on the email account. Then, the user is able to update its profile name :)